common/nodejs-utils/src/config.rs

```diff
@@ -0,0 +1,338 @@
+use crate::error_handling::ErrorType::UserFacing;
+use crate::error_handling::{
+    error_message, ErrorType, SharedErrorMessageBuilder, SuggestRetryBuild, SuggestSubmitIssue,
+};
+use bullet_stream::style;
+use indoc::formatdoc;
+use std::path::Path;
+use std::str::FromStr;
+use toml_edit::TableLike;
+
+const CONFIG_NAMESPACE: [&str; 5] = ["com", "heroku", "buildpacks", "nodejs", "actions"];
+const PRUNE_DEV_DEPENDENCY_CONFIG: &str = "prune_dev_dependencies";
+
+pub fn read_prune_dev_dependencies_from_project_toml(
+    project_toml_path: &Path,
+) -> Result<Option<bool>, ConfigError> {
```

Comment:
[Observation, no action needed] When I see a (non env-var) config, I'll ask if there's something that it can be beyond a boolean. Like if we wanted to have different states beyond on/off for pruning dev dependencies. Like "after_install" or "later" or "force" or something. In this case we don't yet have the abilities to do those things, but if there's a strong chance we might in the future, it's easier to bake it in now, so we don't have to do a config change later. 

I know we're already somewhat down this road with build plan consistency, so I don't think it merits a change now, but while we're looking at it, it's good to consider possible desired use cases (or explicitly state certain usecases are not considered valid or supported).